eno-core/src/main/java/fr/insee/eno/core/processing/in/steps/ddi/DDIMoveUnitInQuestions.java

```diff
@@ -86,6 +89,16 @@ private void applyUnitOnNumericCell(Variable variable, List<Binding> bindings, L
             return;
         }
 
-        numericCell.get().setUnit(variable.getUnit());
+        numericCell.get().setUnit(createUnit(variable.getUnit()));
     }
+
+    public static DynamicLabel createUnit(String value) {
+        if (value == null)
```

Comment:
`if (value == null || value.trim().isEmpty()) {
            throw new InvalidValueException("The value for DynamicLabel cannot be null or empty.");
        }`

Ne serait-ce pas mieux de lever une exception ?

Reply:
Oui je comprends ta remarque c'est le signe d'une faiblesse dans la modélisation de cet objet, mais il a le droit de façon légitime de ne pas avoir de `unit`, dans ces cas là la prop reste à `null` donc pas besoin de créer un objet "label" sur la cible.

Ceci dit, au départ la méthode était en privée ici donc c'était pas trop gênant, mais entre deux je l'ai passée en public pour m'en resservir ailleurs, et dans le "cas général" c'est piégeux que la méthode puisse renvoyer du `null` en effet

Je vais essayer de refacto pour faire plus propre